src/controller/articlesController.js

```diff
@@ -137,6 +137,35 @@ class ArticleController {
     }
   }
 
+  /**
+   *
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @returns {object} - server response
+   */
+  static async getArticleById(req, res) {
+    try {
+      const { articleId } = req.params;
+      const article = await Article.findByPk(articleId);
```

Comment:
A function already exists for finding an article. Please locate it in the middleware section. I will refactor it later to not be a middleware but in order to avoid repetition and reduce code complexity, please use it for now.

Reply:
I think it will reduce the stress during refactoring since it's certain you are going to remove it from the middleware folder

Reply:
Even if I refactor it, it will exist in a function. That way, when I am refactoring, I don't have to look for other instances where it's repeated. I just know that I have to remove the middleware from the route and replace the middleware with the function.

Reply:
Tony, I refactor my code by just creating a helper that updates the user's view stat, I have removed the getAnArticle method